### PR TITLE
 Image cropping function on institution page

### PR DESCRIPTION
**Pull request by Luiz-FS**
* I added a controller to manage the image cropping function.
* I refactored the institution editing controller to crop the image before adding it.
* I refactored the image compression method in the image service to reuse the conversion code from base64 to blob.

### CODE REVIEW

File: app/imageUpload/cropImageController.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
+"use strict";
+
+(function() {
+    var app = angular.module("app");
+
+    app.controller('CropImageController', function CropImageController(image_file, $rootScope, $mdDialog, ImageService, MessageService) {
```
Comment (ruanres) [inline app/imageUpload/cropImageController.js]: quebrar linha

File: app/imageUpload/crop_image.html [hunk 2]
```diff
@@ -0,0 +1,24 @@
+<md-dialog flex flex-gt-md="50" layout="row">
+  <md-card flex="100">
+    <md-card-header>
+      <md-card-header-text>
+        <span class="md-title">Selecione a área que a ser exibida.</span>
```
Comment (ruanres) [inline app/imageUpload/crop_image.html]: Selecione a área a ser exibida.

File: app/imageUpload/cropImageController.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        var cropImgCtrl = this;
+
+        cropImgCtrl.image = '';
+        cropImgCtrl.croppedImage = '';
+
+        function readImage(image_file) {
+            ImageService.readFile(image_file, function (image) {
+                readProperties(image_file, image);
+                $rootScope.$apply(function(){
```
Comment (raonismaneoto) [inline app/imageUpload/cropImageController.js]: Espaço entre () e a {

File: app/imageUpload/crop_image.html (hunk 2 above)
Comment (raonismaneoto) [inline app/imageUpload/crop_image.html]: que será, não?

File: app/imageUpload/cropImageService.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+"use strict";
+
+(function() {
+    var app = angular.module('app');
+
+    app.service("CropImageService", function CropImageService($mdDialog, $q) {
+        var service = this;
+
+        service.cropImage = function cropImage(image_file) {
```
Comment (andreldsa) [inline app/imageUpload/cropImageService.js]: deixa só crop

**Pull request merged**